Code/EnginePlugins/RmlUiPlugin/Resources/RmlUiResource.cpp

```diff
@@ -0,0 +1,151 @@
+#include <RmlUiPluginPCH.h>
+
+#include <Core/Assets/AssetFileHeader.h>
+#include <Foundation/IO/FileSystem/FileReader.h>
+#include <RmlUiPlugin/Resources/RmlUiResource.h>
+
+// clang-format off
+EZ_BEGIN_STATIC_REFLECTED_ENUM(ezRmlUiScaleMode, 1)
+  EZ_ENUM_CONSTANTS(ezRmlUiScaleMode::Fixed, ezRmlUiScaleMode::WithScreenSize)
+EZ_END_STATIC_REFLECTED_ENUM;
+// clang-format on
+
+//////////////////////////////////////////////////////////////////////////
+
+static ezTypeVersion s_RmlUiDescVersion = 1;
+
+ezResult ezRmlUiResourceDescriptor::Save(ezStreamWriter& stream)
+{
+  // write this at the beginning so that the file can be read as an ezDependencyFile
+  m_DependencyFile.StoreCurrentTimeStamp();
+  m_DependencyFile.WriteDependencyFile(stream);
+
+  stream.WriteVersion(s_RmlUiDescVersion);
+
+  stream << m_sRmlFile;
+  stream << m_ScaleMode;
+  stream << m_ReferenceResolution;
+
+  return EZ_SUCCESS;
+}
+
+ezResult ezRmlUiResourceDescriptor::Load(ezStreamReader& stream)
+{
+  m_DependencyFile.ReadDependencyFile(stream);
+
+  ezTypeVersion uiVersion = stream.ReadVersion(s_RmlUiDescVersion);
+
+  stream >> m_sRmlFile;
+  stream >> m_ScaleMode;
+  stream >> m_ReferenceResolution;
+
+  return EZ_SUCCESS;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+// clang-format off
+EZ_BEGIN_DYNAMIC_REFLECTED_TYPE(ezRmlUiResource, 1, ezRTTIDefaultAllocator<ezRmlUiResource>)
+EZ_END_DYNAMIC_REFLECTED_TYPE;
+
+EZ_RESOURCE_IMPLEMENT_COMMON_CODE(ezRmlUiResource);
+// clang-format on
+
+ezRmlUiResource::ezRmlUiResource()
+  : ezResource(DoUpdate::OnAnyThread, 1)
+{
+}
+
+ezResourceLoadDesc ezRmlUiResource::UnloadData(Unload WhatToUnload)
+{
+  ezResourceLoadDesc res;
+  res.m_uiQualityLevelsDiscardable = 0;
+  res.m_uiQualityLevelsLoadable = 0;
+  res.m_State = ezResourceState::Unloaded;
+
+  return res;
+}
+
+ezResourceLoadDesc ezRmlUiResource::UpdateContent(ezStreamReader* Stream)
+{
+  ezRmlUiResourceDescriptor desc;
+  ezResourceLoadDesc res;
+  res.m_uiQualityLevelsDiscardable = 0;
+  res.m_uiQualityLevelsLoadable = 0;
+
+  if (Stream == nullptr)
+  {
+    res.m_State = ezResourceState::LoadedResourceMissing;
+    return res;
+  }
+
+  ezStringBuilder sAbsFilePath;
+  (*Stream) >> sAbsFilePath;
+
+  // Direct loading of rml file
+  if (sAbsFilePath.GetFileExtension() == "rml")
+  {
+    m_sRmlFile = sAbsFilePath;
+
+    res.m_State = ezResourceState::Loaded;
+    return res;
+  }
+
+  ezAssetFileHeader assetHeader;
+  assetHeader.Read(*Stream);
+
+  if (desc.Load(*Stream).Failed())
+  {
+    res.m_State = ezResourceState::LoadedResourceMissing;
+    return res;
+  }
+
+  return CreateResource(std::move(desc));
+}
+
+void ezRmlUiResource::UpdateMemoryUsage(MemoryUsage& out_NewMemoryUsage)
+{
+  // TODO
```

Comment:
TODO

Reply:
Can't really do much better since the resource is just a "description" the majority of memory is used when the resource is intentiated in the component. Will remove the TODO